tests/infra/logging_app.py

```diff
@@ -69,37 +83,39 @@ def issue(
 
         network.wait_for_node_commit_sync()
 
-    def verify(self, network, timeout=3):
+    def verify(self, network=None, node=None, timeout=3):
         LOG.info("Verifying all logging txs")
-
-        remote_node, _ = network.find_primary()
-        for pub_idx, pub_value in self.pub.items():
-            # As public records do not yet handle historical queries,
-            # only verify the latest entry
-            entry = pub_value[-1]
-            self._verify_tx(
-                remote_node,
-                pub_idx,
-                entry["msg"],
-                entry["seqno"],
-                entry["view"],
-                priv=False,
-                timeout=timeout,
-            )
-
-        for priv_idx, priv_value in self.priv.items():
-            for v in priv_value:
+        if network is not None:
+            self.network = network
+        nodes = self.network.get_joined_nodes() if node is None else [node]
```

Comment:
It looks like its possible for `self.network` to still be `None` here, can we add a check and avoid accessing if this is the case?

Reply:
Good point. I've added a check for this.